From aca8018c5b0bbd5435fd6d86a6a49cd071e94e80 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 12:00:16 -0400
Subject: [ethernet] mlx4: Fix wrong dump of the vxlan offloads device capability

Message-id: <5f0e05d1c6d83f8b32d2b64eab6c6f95a0c9191d.1410425016.git.avadai@redhat.com>
Patchwork-id: 93760
O-Subject: [Patch RHEL7.1 V1 080/197] net/mlx4_core: Fix wrong dump of the vxlan offloads device capability
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Or Gerlitz <ogerlitz@mellanox.com>

Fix the value used to dump the vxlan offloads device capability to align
with the MLX4_DEV_CAP_FLAG2_yyy definition. While on that, add dump to
the IPoIB flow-steering device capability and fix small typo.

The vxlan cap value wasn't fully handled when a conflict was resolved
between MLX4_DEV_CAP_FLAG2_DMFS_IPOIB coming from the IB tree to
MLX4_DEV_CAP_FLAG2_VXLAN_OFFLOADS coming from net-next.

Signed-off-by: Or Gerlitz <ogerlitz@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 56cb456746a15c1025a178466492ca4c373b1a63)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/mellanox/mlx4/fw.c b/drivers/net/ethernet/mellanox/mlx4/fw.c
index cb00566..e0b48c6 100644
--- a/drivers/net/ethernet/mellanox/mlx4/fw.c
+++ b/drivers/net/ethernet/mellanox/mlx4/fw.c
@@ -129,13 +129,14 @@ static void dump_dev_cap_flags2(struct mlx4_dev *dev, u64 flags)
   [0] = "RSS support",
   [1] = "RSS Toeplitz Hash Function support",
   [2] = "RSS XOR Hash Function support",
-  [3] = "Device manage flow steering support",
+  [3] = "Device managed flow steering support",
   [4] = "Automatic MAC reassignment support",
   [5] = "Time stamping support",
   [6] = "VST (control vlan insertion/stripping) support",
   [7] = "FSM (MAC anti-spoofing) support",
   [8] = "Dynamic QP updates support",
-  [9] = "TCP/IP offloads/flow-steering for VXLAN support"
+  [9] = "Device managed flow steering IPoIB support",
+  [10] = "TCP/IP offloads/flow-steering for VXLAN support"
  };
  int i;
 
-- 
1.7.1